### ISSUE TITLE
Understand what tf.nn.log_softmax and tf.multinomial does

### ISSUE THREAD

**Issue by vaibhav369** [vaibhav369/Reinforcement-Learning#2]
More specifically, why should tf.nn.log_softmax be used in Vanilla Policy Gradient, when mathematical formula shows simple log. Why does tf.nn.log fail?
